quetz/cli.py

```diff
@@ -249,76 +258,17 @@ def _fill_test_database(db: Session) -> NoReturn:
         db.close()
 
 
-def _get_deployments() -> Dict[str, str]:
-    """Get a mapping of the current Quetz deployments.
-
-    Returns
-    -------
-    deployments : Dict[str, str]
-        The mapping of deployments
-    """
-
-    if os.path.exists(_deployments_file):
-        return _get_cleaned_deployments()
-    else:
-        Path(_user_dir).mkdir(parents=True, exist_ok=True)
-        return {}
-
-
-def _store_deployment(path: str, config_file_name: str) -> NoReturn:
-    """Store a new Quetz deployment.
-
-    Parameters
-    ----------
-    path : str
-        The location of the deployment
-    config_file_name : str
-        The configuration file name, including its extension
-    """
-
-    json_ = {path: config_file_name}
-    deployments = _get_deployments()
-
-    deployments.update(json_)
-    with open(_deployments_file, 'w') as f:
-        json.dump(deployments, f)
-
-
-def _get_cleaned_deployments() -> Dict[str, str]:
-    """Get a cleaned version of deployments.
-
-    This could be necessary to clean-up if the user delete manually a deployment
-    directory without updating the deployments files in its profile.
-
-    Returns
-    -------
-    deployments : Dict[str, str]
-        The mapping of deployments
-    """
-
-    with open(_deployments_file, 'r') as fid:
-        deployments: Dict[str, str] = json.load(fid)
-
-    to_delete = []
-    for path, f in deployments.items():
-        config_file = os.path.join(path, f)
-        if not os.path.exists(config_file):  # User has deleted the instance without CLI
-            to_delete.append(path)
-
-    cleaned_deployments = {
-        path: f for path, f in deployments.items() if path not in to_delete
-    }
-
-    if len(to_delete) > 0:
-        with open(_deployments_file, 'w') as fid:
-            json.dump(cleaned_deployments, fid)
-
-    return cleaned_deployments
-
-
-def _clean_deployments():
-    """Clean the deployments without returning anything."""
-    _ = _get_cleaned_deployments()
+def _is_deployment(base_dir: Path):
+    config_file = base_dir.joinpath("config.toml")
+    if (
+        base_dir.exists()
+        and config_file.exists()
+        and base_dir.joinpath("channels").exists()
+    ):
+        config = Config(str(config_file.resolve()))
+        with working_directory(base_dir):
+            return database_exists(config.sqlalchemy_database_url)
```

Comment:
this may be misleading if the user has configured the pg db url, but did not create the database yet (or had forgotten to do so).

Reply:
you're right.
What are the minimal checks I should perform?

Reply:
maybe we could just skip this check and raise an error at runtime if the db does not exist. What do you think @wolfv?

Reply:
I agree, a runtime error is fine with me